app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,66 @@
+class VendorsController < ApplicationController
+  # (30*24*60*60) = 2592000
+  ONE_MONTH = 2592000
+
+  def index
+    @vendors = Vendor.all
+    @main_page = true
+  end
+
+  def show
+    @vendor = Vendor.find(params[:id])
+    @vendor ||= Vendor.find(params[:vendor_id])
```

Comment:
this should probably be rewritten as:
`@vendor = Vendor.find(params[:id]) || Vendor.find(params[:vendor_id])`